cmd/rpcdaemon/commands/zkevm_api.go

```diff
@@ -382,6 +400,96 @@ func (api *ZkEvmAPIImpl) GetBatchWitness(ctx context.Context, batchNumber uint64
 	return api.getBlockRangeWitness(ctx, api.db, startBlock, endBlock, false)
 }
 
+func (api *ZkEvmAPIImpl) GetAggregatorDetail(ctx context.Context, batchNumber uint64, debug *bool) (*legacy_executor_verifier.RpcPayload, error) {
+	if !sequencer.IsSequencer() {
+		return nil, errors.New("method only supported from a sequencer node")
+	}
+
+	useDebug := false
+	if debug != nil {
+		useDebug = *debug
+	}
+
+	tx, err := api.db.BeginRo(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer tx.Rollback()
+
+	chainConfig, err := api.ethApi.chainConfig(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	hDb := hermez_db.NewHermezDbReader(tx)
+
+	blockNumbers, err := hDb.GetL2BlockNosByBatch(batchNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	lastBlock, err := rawdb.ReadBlockByNumber(tx, blockNumbers[len(blockNumbers)-1])
+	if err != nil {
+		return nil, err
+	}
+
+	start := rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(blockNumbers[0]))
+	end := rpc.BlockNumberOrHashWithNumber(rpc.BlockNumber(blockNumbers[len(blockNumbers)-1]))
+
+	streamBytes, err := getStreamBytes(tx, batchNumber, blockNumbers, lastBlock, hDb, chainConfig.ChainID.Uint64())
+	if err != nil {
+
+	}
+
+	rangeWitness, err := api.getBlockRangeWitness(ctx, api.db, start, end, useDebug)
+	if err != nil {
+		return nil, err
+	}
+
+	oldAccInputHash, err := api.l1Syncer.GetOldAccInputHash(ctx, &api.config.L1PolygonRollupManager, ApiRollupId, batchNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	timestampLimit := lastBlock.Time()
+
+	return &legacy_executor_verifier.RpcPayload{
+		Witness:           hex.EncodeToHex(rangeWitness),
+		DataStream:        hex.EncodeToHex(streamBytes),
```

Comment:
I am thinking that that one is not really necessary, they can get the datastream from our datastream server.